### PR TITLE
refactor: Move src/interfaces/*.cpp files to libbitcoin_common.a

### PR DESCRIPTION
**Pull request by ryanofsky**
These belong in `libbitcoin_common.a`, not `libbitcoin_util.a`, because they aren't general-purpose utilities, they just contain some common glue code that is used by both the node and the wallet. Another reason not to include these in `libbitcoin_util.a` is to prevent them from being used by the kernel library.

Also rename ambiguous `MakeHandler` functions to `MakeCleanupHandler` and `MakeSignalHandler`. Cleanup function handler was introduced after boost signals handler, so original naming didn't make much sense.

This just contains a move-only commit, and a rename commit. There are no actual code or behavior changes.

This PR is an alternative to #26293, and solves the same issue of removing a boost dependency from the _util_ library. The advantages of this PR compared to #26293 are that it keeps the source directory structure more flat, and it avoids having to change #includes all over the codebase.

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK - closed #26293 in favour of this PR.

> and solves the same issue of removing a boost dependency from the util library.

Want to pull in https://github.com/bitcoin/bitcoin/pull/26293/commits/c30a390760e3fe0af00d161fccfcdc93ae55f9dc or make a similar change here?

**Comment by ryanofsky** [comment]
Updated ba4f6c501bf277ebfbeaec57ed25e65b8764d017 -> 879042a114fa60937ed9732d82986a17361a7dcf ([`pr/intclean.1`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.1) -> [`pr/intclean.2`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/intclean.1..pr/intclean.2)) adding fanquake's commit dropping boost dependency from util library

> Want to pull in [c30a390](https://github.com/bitcoin/bitcoin/commit/c30a390760e3fe0af00d161fccfcdc93ae55f9dc) or make a similar change here?

Thanks, done now

**Comment by hebasto** [comment]
Concept ACK.

**Comment by ryanofsky** [comment]
Rebased 879042a114fa60937ed9732d82986a17361a7dcf -> 8e5ab1b6433629ee2f5cf29c5ae24fd61155006c ([`pr/intclean.2`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.2) -> [`pr/intclean.3`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.3), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/intclean.2-rebase..pr/intclean.3)) due to conflict with #26313

File: src/interfaces/init.h [hunk 1]
```diff
@@ -5,6 +5,12 @@
 #ifndef BITCOIN_INTERFACES_INIT_H
 #define BITCOIN_INTERFACES_INIT_H
 
+#include <interfaces/chain.h>
+#include <interfaces/echo.h>
+#include <interfaces/init.h>
```
Comment (hebasto) [inline src/interfaces/init.h:10]: 0087e2e36ddeec15adfce7cd72579a9a73e37540

Why this header?

File: src/interfaces/init.h [hunk 2]
```diff
@@ -29,11 +35,11 @@ class Init
...
-    virtual std::unique_ptr<Node> makeNode();
-    virtual std::unique_ptr<Chain> makeChain();
-    virtual std::unique_ptr<WalletLoader> makeWalletLoader(Chain& chain);
-    virtual std::unique_ptr<Echo> makeEcho();
-    virtual Ipc* ipc();
+    virtual std::unique_ptr<Node> makeNode() { return nullptr; }
+    virtual std::unique_ptr<Chain> makeChain() { return nullptr; }
+    virtual std::unique_ptr<WalletLoader> makeWalletLoader(Chain& chain) { return nullptr; }
+    virtual std::unique_ptr<Echo> makeEcho() { return nullptr; }
+    virtual Ipc* ipc() { return nullptr; }
```
Comment (hebasto) [inline src/interfaces/init.h:42]: 0087e2e36ddeec15adfce7cd72579a9a73e37540

Add `#include <interfaces/ipc.h>`?

**Review by ryanofsky — COMMENTED** [review]
Thanks for the review!

Updated 8e5ab1b6433629ee2f5cf29c5ae24fd61155006c -> 9b093fdacfd3a4b82ad2024edcb5f8ec24a55d42 ([`pr/intclean.3`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.3) -> [`pr/intclean.4`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.4), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/intclean.3..pr/intclean.4)) with suggested cleanups

**Review by hebasto — COMMENTED** [review]
Approach ACK 9b093fdacfd3a4b82ad2024edcb5f8ec24a55d42.

File: src/common/interfaces.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <interfaces/echo.h>
+#include <interfaces/handler.h>
+
+#include <boost/signals2/connection.hpp>
+#include <utility>
+#include <memory>
```
Comment (hebasto) [inline src/common/interfaces.cpp:10]: pico-nit: `clang-format-diff.py` complains about header order:
```suggestion
#include <memory>
#include <utility>
```

**Comment by hebasto** [comment]
This PR needs to be built on top of 39adc3e90232f6e9aaf9d484007723a9136f0df0 from #26294. At least, on macOS M1.

**Comment by hebasto** [comment]
> This PR needs to be built on top of [39adc3e](https://github.com/bitcoin/bitcoin/commit/39adc3e90232f6e9aaf9d484007723a9136f0df0) from #26294. At least, on macOS M1.

As #26294 has been merged, mind rebasing this PR?

File: src/common/interfaces.cpp (hunk 3 above)
Comment (ryanofsky) [reply src/common/interfaces.cpp]: > pico-nit: `clang-format-diff.py` complains about header order:

Thanks, fixed

**Review by ryanofsky — COMMENTED** [review]
> As #26294 has been merged, mind rebasing this PR?

Thanks, rebased now

---

Rebased 9b093fdacfd3a4b82ad2024edcb5f8ec24a55d42 -> b19c4124b3c9a15fe3baf8792c55eb26eca51c0f ([`pr/intclean.4`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.4) -> [`pr/intclean.5`](https://github.com/ryanofsky/bitcoin/commits/pr/intclean.5), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/intclean.4-rebase..pr/intclean.5)) after #26294 to fix macos build issue

**Review by hebasto — APPROVED** [review]
ACK b19c4124b3c9a15fe3baf8792c55eb26eca51c0f

**Pull request merged**